### DISCUSSION TITLE
Event allowing overriding card HTML displayed in the control panel

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#15287] [Ideas] [answered]
Currently card contents are defined by selecting specific fields to be displayed in it - in field layout designer.
Is there an option to just override Card html with some event?
If there is not, it should be.

**Comment by mmikkel** [comment, marked as answer]
There is a [`Cp::EVENT_DEFINE_ELEMENT_CARD_HTML`](https://docs.craftcms.com/api/v5/craft-helpers-cp.html#event-define-element-card-html) event, which can be used to manipulate the HTML for element cards inside the control panel. I actually [use this event](https://github.com/mmikkel/CpFieldInspect-Craft/blob/327b10dd348ab0908d61c7702acc0f13ba0e019e/src/CpFieldInspect.php#L113) in my CP Field Inspect plugin, in order to add a "Edit Entry Type" link to card action menus.

However, if the goal is to add some custom content to the card in addition to the custom fields exposed via the "Show in element cards" layout element option in field layout designers, that event is a bit cumbersome to use, since you have to parse the entire HTML for the rendered card and figure out how and where to inject your custom content.

To that end, it'd be great if another event (like `EVENT_DEFINE_ELEMENT_CARD_BODY_FIELDS` or similar) could be added, that could be used to specifically add custom content to cards – and/or possibly even manipulate how existing card body fields are presented (sort of like how the `EVENT_DEFINE_ATTRIBUTE_HTML` event works for element indexes).

**Reply by brandonkelly** [reply]
We’re adding an `EVENT_REGISTER_CARD_ATTRIBUTES` event via #15283 ;)
